apps/www/_blog/2022-12-14-mfa-auth-via-rls.mdx

```diff
@@ -0,0 +1,149 @@
+
+
+## An Overview of TOTP
+
+[TOTP](https://www.rfc-editor.org/rfc/rfc6238) works by generating a unique, one-time password that is valid for a limited amount of time, usually 30 seconds or less. This password is generated using a shared secret key that is known only to the device and Supabase Auth, along with the current time. To exchange the shared secret key, a user scans a QR code generated by the server in order to establish a connection.  The QR code can be represented by a URI which conforms to the [Google Authenticator Key URI format](https://github.com/google/google-authenticator/wiki/Key-Uri-Format):
+
+```jsx
+otpauth://totp/supabase.io:j@supacats.io?algorithm=SHA1&digits=6&issuer=supabase.io&period=30&secret=BFSXQHFB2BGAZIOQWCDBJUF7B54A52JQ
+```
+
+The first portion of  `otpauth://totp/supabase.io` describes the protocol and issuer while `j@supacats.io` refers to the user. The remaining parameters refer to specifics around OTP generation. In this case, the OTP code is generated using a SHA1 hash of the secret combined with the timestamp and the OTP code is valid for `30s`
+
+In the event that the user faces difficulties entering a QR code the user can also opt to manually type the secret into the authenticator device.
+
+## MFA Flows: Enrolment and Verification
+
+An MFA flow can be broken into two key steps: Enrollment and Verification. During the *Enrollment* process Supabase Auth exchanges secrets with the user’s authenticator application. During the *Verification* process, the device makes use of the timestamp together with the secret to produce a six digit code that the server can verify.
+
+**Enrollment**
+
+To generate a QR code, call the `/enroll` endpoint which returns an SVG encoded QR and the secret. Thereafter, create a challenge by calling the `/challenge` endpoint. Once the user has entered the six digit TOTP code generated by their authenticator app, call the`/verify`  endpoint with the corresponding factor and challenge details.
+
+You might wonder: why the need for the challenge step? The challenge step is designed to create an interval between MFA initiation and the action of making a verification. This is useful in cases like Yubikey authentication where a user might need to request a challenge before placing their finger on the device.
+
+```mermaid
+graph TD
+    Z[Start] --> A
+    A[Enroll Factor] --> B(Challenge Factor) -->  K(Verify Factor)
+    K -->|Correct Code| D[Factor Verified, Logged in ]
+    K -->|Incorrect Code/Error| Z
+
+```
```

Comment:
Can we verify that `mermaid` is supported _before_ we publish the post?

Reply:
Looks like it isn't :( Will go with adding images unless the mermaid-mdx works fine and frontend approves of it